File: Week4_Arrays/remove-duplicates-easy.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
        
        uniquePtr = 0
        for i in range(1, len(nums)):
            if nums[i] != nums[i - 1]:
                uniquePtr += 1
                nums[uniquePtr] = nums[i]
        
        return uniquePtr + 1

File: Week4_Arrays/remove-duplicates-easy.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums.length == 0) {
            return 0;
        }

        int uniquePtr = 0;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] != nums[i - 1]) {
                uniquePtr++;
                nums[uniquePtr] = nums[i];
            }
        }

        return uniquePtr + 1;
    }
}